crates/bevy_color/src/color_range.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{LinearRgba, Mix};
+
+/// Represents a range of colors that can be linearly interpolated, defined by a start and
+/// end point which must be in the same color space. It works for any color type that
+/// implements [`Mix`].
+///
+/// This is useful for defining gradients or animated color transitions.
+pub struct ColorRange<T: Mix> {
+    start: T,
+    end: T,
+}
+
+impl<T> ColorRange<T>
+where
+    T: Mix,
+{
+    /// Construct a new color range from the start and end values.
+    pub fn new(start: T, end: T) -> Self {
+        Self { start, end }
+    }
+
+    /// Get the color value at the given interpolation factor, which should be between 0.0
+    /// and 1.0.
+    pub fn at(&self, factor: f32) -> T {
+        self.start.mix(&self.end, factor)
+    }
+}
+
+/// A type-erased color range that can be used to interpolate between colors in various
```

Comment:
The docs aren't sufficiently clear on why this is needed, how it can be used, or how it relates to `ColorRange`.

Reply:
Expanded comment.